### ISSUE TITLE
[Bug]: Error reconnecting to network after leaving

### ISSUE THREAD

**Issue by bmcgonag** [gravitl/netmaker#1111]
### Contact Details

brian@fixitdelrio.com

### What happened?

I connected to the 0.14.0 server I created on Digital Ocean with the quick install (the bash script), and it seemed fine. Created a new network, and then an access key for that new network. 

I used the netclient from a Ubuntu 22.04 desktop and connected (still unable to find the GUI client BTW, but neither here nor there).  I was cut off from the rest of my netowrk and had no internet access once connected, so I left the network using the netclient command `sudo netclient leave -n <network name>`.

I was then able to get to the internet again, and added an egress gateway to the network I created (the server as the node). 
I then attempted to reconnect using the same command I did the  first time, and I get an error. Also, I make it so I had about 20 uses of the access key I created, and had only used it once.

The error is as follows:

```
$ sudo netclient join -t my-super-long-key-that-netmaker-generated==
[netclient] 2022-05-21 14:52:10 joining DRSP at api.nm.142-93-247-25.nip.io:443 
[netclient] 2022-05-21 14:52:10 unable to authenticate: failed to authenticate 400 Bad Request {"Code":400,"Message":"W1R3: MacAddress can't be empty"} 
[netclient] 2022-05-21 14:52:12 removed systemd remnants if any existed 
[netclient] 2022-05-21 14:52:13 removed systemd remnants if any existed
```

### Version

v0.14.0

### What OS are you using?

Linux

### Relevant log output

```shell
$ sudo netclient join -t my-super-long-key-that-netmaker-generated==
[netclient] 2022-05-21 14:52:10 joining DRSP at api.nm.142-93-247-25.nip.io:443 
[netclient] 2022-05-21 14:52:10 unable to authenticate: failed to authenticate 400 Bad Request {"Code":400,"Message":"W1R3: MacAddress can't be empty"} 
[netclient] 2022-05-21 14:52:12 removed systemd remnants if any existed 
[netclient] 2022-05-21 14:52:13 removed systemd remnants if any existed
```

### Contributing guidelines

- [X] Yes, I did.

**Comment by mattkasun** [comment]
Check if the node is still displayed in the Ui and if so, delete it from the server before trying to rejoin. 

Linux packages do not include the GUI. You can download the linux-gui from the releases

**Comment by bmcgonag** [comment]
Thanks for the reply. Can you point me to the correct linux gui download. I see nothing in the 0.14.0 releases about a gui.  All say netclient-xxx or just netclient.

**Comment by mattkasun** [comment]
It appears the workflow failed to upload the GUI version.

**Comment by dschaper** [comment]
I have the same situation, deleted the client from the admin interface and now the client is unauthorized. 

```plain
# netclient join -t <token> -vvv
[netclient] 2022-05-22 17:01:44 joining <redact>vpn at api.<redact>:443
[netclient] 2022-05-22 17:01:44 error installing:  error creating node 401 Unauthorized {"Code":401,"Message":"You are unauthorized to access this endpoint."}
[netclient] 2022-05-22 17:01:44 unable to authenticate: failed to authenticate 400 Bad Request {"Code":400,"Message":"W1R3: MacAddress can't be empty"}
[netclient] 2022-05-22 17:01:44 could not flush peer routes when leaving network,  <redact>vpn
[netclient] 2022-05-22 17:01:44 removed  <redact>vpn  network locally
[netclient] 2022-05-22 17:01:46 removed systemd remnants if any existed
[netclient] 2022-05-22 17:01:47 removed systemd remnants if any existed
```

**Comment by afeiszli** [comment]
fixed in 0.14.1: removed MacAddress limitation on join and authorization
